### PR TITLE
/repopulate <chunk x> <chunk z> command and some rng tools for gateway generation

### PR DESCRIPTION
**Pull request by Robitobi01**
Xcom wanted to add a command for chunk deletion but never did till now, most of the time re-population is probably enough. So now there's a command for that.

### CODE REVIEW

File: carpetmodSrc/carpet/commands/CommandRNG.java [hunk 1]
```diff
@@ -65,6 +67,7 @@ public String getUsage(ICommandSender sender) {
      */
     public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException {
         if (!command_enabled("commandRNG", sender)) return;
+        if (args.length <=0) return;
```
Comment (Earthcomputer) [inline carpetmodSrc/carpet/commands/CommandRNG.java]:
```suggestion
        if (args.length <=0) throw new WrongUsageException(getUsage(sender));
```
(don't merge via github UI, it needs an import too :P)

**Comment by Robitobi01** [comment]
A lot of the /rng command ignores argument length, I added a base check for length > 0 because just sending /rng without arguments already throws ArrayIndexOutOfBoundsExceptions. Such a fix is required/missing for all subcommands too....

File: carpetmodSrc/carpet/commands/CommandRNG.java (hunk 1 above)
Comment (Robitobi01) [reply carpetmodSrc/carpet/commands/CommandRNG.java]: Added, eventhough afterwards I noticed it's already imported for catching errors in my own additions xD

**Review by Earthcomputer — APPROVED** [review]

File: carpetmodSrc/carpet/commands/CommandRepopulate.java [hunk 2]
```diff
@@ -43,4 +48,18 @@ public void execute(MinecraftServer server, ICommandSender sender, String[] args
         }
     }
 
+    @Override
+    public List<String> getTabCompletions(MinecraftServer server, ICommandSender sender, String[] args, @Nullable BlockPos targetPos) {
+        int chunkX = sender.getPosition().getX() >> 4;
+        int chunkZ = sender.getPosition().getZ() >> 4;
+
+        if (args.length == 1) {
+            return getListOfStringsMatchingLastWord(args, Integer.toString(chunkX));
```
Comment (Earthcomputer) [inline carpetmodSrc/carpet/commands/CommandRepopulate.java:57]: This isn't the normal way of completing a location. Take a look at how it's done in the `/fillbiome` command.

Comment (Robitobi01) [reply carpetmodSrc/carpet/commands/CommandRepopulate.java:57]: I'm aware of that, but for chunk coordinates using looking at makes not much sense and when flying somewhere in the sky you can just tab complete the chunk and it doesn't require you to have a block in place. The last ten times I used it while testing I just flew to the chunk and saved the tab completed command I can execute further away.
Adding ~ might be an option but then u need to take care of relative chunk coords. Could also change it to block coords and convert it down to chunk coordinates internally.
This is just the most intuitive way imo.

**Pull request merged**